pkg/model/task.go

```diff
@@ -129,6 +129,7 @@ func parseStorageSource(path string, resource *Resource) StorageSpec {
 
 func parseInputs(mounts IPLDMap[string, Resource]) ([]StorageSpec, error) {
 	inputs := []StorageSpec{}
+	// TODO(forrest): [correctness] this is correct but easily bug prone: https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable
```

Comment:
I think this whole function could be 

```go
return lo.MapToSlice(mounts.Values, func(path string, resource Resource) StorageSpec {
    return parseStorageSource(path, &resource)
})
```
if parseStorageSource didn't require a pointer parameter, it could be 

```go
return lo.MapToSlice(mounts.Values, parseStorageSource)
```